File: lib/bindings/tests/bindings/test_breez_sdk_liquid.swift
import breez_sdk_liquid

class SDKListener: EventListener {
    func onEvent(e: SdkEvent) {
        print("Received event ", e);
    }
}

let mnemonic = "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about";
let config = breez_sdk_liquid.defaultConfig(network: .testnet, breezApiKey: nil);
let connectRequest = breez_sdk_liquid.ConnectRequest(config: config, mnemonic: mnemonic);
let sdk = try breez_sdk_liquid.connect(req: connectRequest);

let listenerId = try sdk.addEventListener(listener: SDKListener());

let nodeInfo = try sdk.getInfo();

try sdk.removeEventListener(id: listenerId);

print(nodeInfo);
assert(nodeInfo.pubkey == "03d902f35f560e0470c63313c7369168d9d7df2d49bf295fd9fb7cb109ccee0494", "nodeInfo.pubkey");

File: lib/bindings/tests/bindings/test_breez_sdk_liquid.py
import breez_sdk_liquid


class SDKListener(breez_sdk_liquid.EventListener):
   def on_event(self, event):
      print(event)


def test():
    mnemonic = "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
    config = breez_sdk_liquid.default_config(breez_sdk_liquid.LiquidNetwork.TESTNET, None)
    config.sync_service_url = None
    connect_request = breez_sdk_liquid.ConnectRequest(config=config, mnemonic=mnemonic)
    sdk = breez_sdk_liquid.connect(req=connect_request)

    listener_id = sdk.add_event_listener(SDKListener())

    node_info = sdk.get_info()

    sdk.remove_event_listener(listener_id)

    print(node_info)
    assert node_info.wallet_info.pubkey == "03d902f35f560e0470c63313c7369168d9d7df2d49bf295fd9fb7cb109ccee0494"

test()
